### PR TITLE
New merge datasets footer summary

### LINKED ISSUE #4610: Add footer summary on merge tables
We're replacing the connection line to a small "puzzle piece" on the footer with a summary of the opperation:

screen shot 2015-07-21 at 23 12 12
screen shot 2015-07-22 at 01 05 57
screen shot 2015-07-22 at 01 05 11

Last version on the sketch file you already have. I'll help you tomorrow with this:+1:

### PR DESCRIPTION
**Pull request by viddo**
Fixes #4610

Basically, introduces the new style "Puzzle" concept, that replaces the "connecting lines" ornaments. 

Column merge;
column1-1-no-selections
column1-2-selected-left
column1-3-both-selections
column2-1

Spatial merge:
spatial1-1-no-footer-info
spatial2-1-no-selected
spatial2-2-count
spatial2-3-missing-merge-method
spatial2-4-merge

@xavijam please code review, extracted the footer view to be reusable and rendered through each step instead. Each step has it's own summary (footer info) that contains the specifics.

@josecruz @saleiva for visuals. Also, shouldn't we remove the sticky header, considering we now have the summary in the footer? See: column2-2-sticky

### CODE REVIEW

**Comment by viddo** [comment]
retest this please

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/5267/))

File: lib/assets/javascripts/cartodb/common/dialogs/merge_datasets/spatial_merge/footer_info.jst.ejs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<div class="Puzzle">
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb/common/dialogs/merge_datasets/spatial_merge/footer_info.jst.ejs]: go puzzle go!

**Comment by xavijam** [comment]
:jp: puzzle concept, i love it.

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/5278/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/5289/))

**Comment by viddo** [comment]
retest this please

**Comment by josecruz** [comment]
Found this: 
screen shot 2015-07-23 at 1 20 23 pm

Also, on the sticked header, we could replace with the dataset name, @saleiva what do you think?

And on the `count` puzzle piece we could show the name of both dataset, kind of: 

dataset_a  ∩  dataset_b

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/5299/))

**Comment by viddo** [comment]
The cut text is fixed, the assets on the staging servers was not up-to-date when you checked propably, redeploying now.

Regarding what to set in the pieces, please think through how you want it for all the different states (column vs. spatial, step 1 or 2, having selected items or not etc.), and take into account the sticky header, and let me know what you want and I'll fix it.

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/5302/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/5333/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/5339/))

**Comment by viddo** [comment]
Last changed based on info I got offline; for count merge method we display the right table name instead of a column name: 
screen shot 2015-07-24 at 10 29 39

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/5341/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/5342/))

**Comment by josecruz** [comment]
Yay! :+1:

**Pull request merged**